src/Repositories/ConsultationRepository.php

```diff
@@ -33,4 +29,15 @@ public function allQueryBuilder(array $search = [], array $criteria = []): Build
         }
         return $query;
     }
+
+    public function updateModel(Consultation $consultation, array $data): Consultation
+    {
+        foreach ($data as $k => $v) {
+            if (isset($consultation->$k)) {
```

Comment:
I don't understand what you are trying to do here, `isset` will return false on `null`, so if Consultation has some existing fields set to null they can never be updated...
You probably should remove all this code and just do `$consultation->fill($data)` which will update only `$fillable` fields.